src/components/Popup/Popup.tsx

```diff
@@ -245,17 +254,23 @@ export default class Popup extends AutoControlledComponent<Extendable<PopupProps
     rtl: boolean,
     accessibility: AccessibilityBehavior,
   ): JSX.Element {
-    const { align, position } = this.props
+    const { align, position, offset } = this.props
     const { target } = this.state
 
     const placement = computePopupPlacement({ align, position, rtl })
 
+    const popperModifiers = {
+      // https://popper.js.org/popper-documentation.html#modifiers..offset
+      ...(offset && { offset: { offset: this.applyRtlToOffset(offset, rtl, position) } }),
```

Comment:
Too bad applying modifiers is the only way we can adjust the offset... There is no way we can use this prop inside the styles in order to achieve the position, right? It really feels awkward to have to tackle rtl in the component's logic..

Reply:
yes, there is no any - so, there are several reasons for that. 

First is simple - there are no additional props of Popper that would allow client to customize positioning (actually, good thing). And here we are getting to the next alternative option that might come to mind (the one you've mentioned) - use styles for influencing these offsets.

I've tried this one - and have done that only to clearly see that even in LTR mode taken in isolation we have lots of issues. As an example: if it is about 'start' value for `align` property, which corresponds to popup aligned with the left side of trigger, 

![image](https://user-images.githubusercontent.com/9024564/49904105-756e8a80-fe69-11e8-9fa5-9e8efada759b.png)

we might use `left` to specify an offset (as `absolute` positioning strategy is used by Popper). However, when we will achieve necessary result and switch to the ride sides alignment, we will see that Popper is using `transform` property to ensure alignment for the right sides, and here we cannot use offsets of `left-right` CSS properties anymore.

For the sake of not preventing havoc to wreak our code, decided to just reuse solution Popper has already introduced for solving exactly this problem (probably, the most reasonable thing) - and this, in fact, what we've done with the modifiers. Note, we are not abusing it - we are using the tool from Popper that is, actually, dedicated to solve this exact problem :)